py/misc.h

```diff
@@ -33,6 +33,8 @@
 #include <stdbool.h>
 #include <stdint.h>
 #include <stddef.h>
+// Needed for NORETURN
+#include "py/mpconfig.h"
```

Comment:
Don't think this is explicitly documented, but see the rest of the code (grep for all occurrences of misch.h): the idea is that any file needing to include py/misc.h first includes py/mpconfig.h, so this change should not be made.